src/tests/ftest/util/verify_perms.py

```diff
@@ -260,12 +260,10 @@ def _real_w(entry_type, path):
     '''
     if entry_type == 'file':
         try:
-            # Write a newline to reduce likelihood of corrupting an executable file
+            # Always write a line that allows the file to be executable
             with open(path, "w", encoding='utf-8') as file:
-                # O_APPEND is not supported by DFS, so "a" is not used in open. Need seek() here
-                # to move file pointer to the end of the file.
-                file.seek(0, 2)
-                return file.write('\n') == 1
+                data = '#!/usr/bin/env bash\n'
+                return file.write(data) == len(data)
```

Comment:
This change was cherry-picked from https://github.com/daos-stack/daos/commit/b33a13218078f7bcdcb0032e9927b81cd55ba3be
It should merge cleanly now